File: Napakalaki-Java/src/NapakalakiGame/BadConsequence.java

package NapakalakiGame;

import java.util.ArrayList;

public abstract class BadConsequence {
    static final int MAXTREASURES=10;
    protected String text;
    protected int levels;
 
    public BadConsequence(String text_i,int lvl){
        text = text_i;
        levels = lvl;
    }  
    public String getText(){
        return text;
    }
    public int getLevels(){
        return levels;
    }
    public abstract boolean isEmpty();
    abstract void substractVisibleTreasure(Treasure t);
    abstract void substractHiddenTreasure(Treasure t);
    abstract BadConsequence adjustToFitTreasureLists(ArrayList<Treasure> v, ArrayList<Treasure> h);
    public String toString(){
        String out;
        out =  "Text = " + text + ", Levels = " + Integer.toString(levels);
        return out;
    }



//    public boolean isEmpty(){
//        boolean empty=false;
//       if(nVisibleTreasures==-1 && nHiddenTreasures==-1)
//           empty = specificHiddenTreasures.isEmpty() && specificHiddenTreasures.isEmpty();
//       else
//               if(nVisibleTreasures==0 && nHiddenTreasures==0)
//                   empty=true;
//       return empty;
//          
//    }
    
//    void substractVisibleTreasure(Treasure t){
//        if(nVisibleTreasures==-1)
//            specificVisibleTreasures.remove(t.getType());    
//        else
//            if(nVisibleTreasures!=0)
//                nVisibleTreasures--;
//    }
    
//    void substractHiddenTreasure(Treasure t){
//        if(nHiddenTreasures==-1)
//            specificHiddenTreasures.remove(t.getType());    
//        else
//            if(nHiddenTreasures!=0)
//                nHiddenTreasures--;
//    }
    
//    BadConsequence adjustToFitTreasureLists(ArrayList<Treasure> v, ArrayList<Treasure> h){
//       BadConsequence out;
//       if(!death){
//           if(nVisibleTreasures!=-1 && nHiddenTreasures!=-1){
//               int nuevoNV,nuevoNH;
//               nuevoNV=nVisibleTreasures>v.size()?v.size():nVisibleTreasures;
//               nuevoNH=nHiddenTreasures>h.size()?h.size():nHiddenTreasures;
//               out= new BadConsequence(text,levels,nuevoNV,nuevoNH);
//           }
//           else{
//               ArrayList<TreasureKind> nuevoSV= new ArrayList(0);
//               ArrayList<TreasureKind> copiaVT= new ArrayList(0);
//               ArrayList<TreasureKind> copiaHT = new ArrayList(0);
//               int index=0;
//               for(Treasure treasure:v){
//                    copiaVT.add(treasure.getType());
//               }
//               for(Treasure treasure:h){
//                    copiaHT.add(treasure.getType());
//               }
//               for(TreasureKind tipo:specificVisibleTreasures){
//                   if((index=copiaVT.indexOf(tipo))!=-1){
//                       nuevoSV.add(tipo);
//                       copiaVT.remove(index);
//                   }
//               }
//               ArrayList<TreasureKind> nuevoSH= new ArrayList(0);
//                for(TreasureKind tipo:specificHiddenTreasures){
//                   if((index=copiaHT.indexOf(tipo))!=-1){
//                       nuevoSH.add(tipo);
//                       copiaHT.remove(index);
//                   }
//                }
//               out=new BadConsequence(text,levels,nuevoSV,nuevoSH);
//           }  
//       }
//       else{
//           out= new BadConsequence(text,levels,v.size(),h.size());
//       }
//       return out;
//    }
    
 
//    public String toString(){
//        String out;
//        out =  "Text = " + text;
//        if(death)
//            out += ", death = " + Boolean.toString(death);
//        else{
//            if(specificHiddenTreasures!=null){
//                 out += ", Hidden Treasures = ";
//                for (TreasureKind specificHT : specificHiddenTreasures) {
//                    out += ", " + specificHT.toString();
//                }
//            }
//            if(specificVisibleTreasures!=null){
//                out += ", Visible Treasures = ";
//                for (TreasureKind specificVT : specificVisibleTreasures) {
//                out += ", " + specificVT.toString();
//                }
//            }
//            if(nVisibleTreasures>=0)
//                out += ", number of Visible Treasures = " + Integer.toString(nVisibleTreasures);
//            if(nHiddenTreasures>=0)
//                out += ", number of Hidden Treasures = " + Integer.toString(nHiddenTreasures);
//            if(levels>=0)
//                out += ", Levels = " + Integer.toString(levels);
//        }
//        return out;
//    }
    
}

File: Napakalaki-Ruby/lib/bad_consequence.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame
class BadConsequence
  def initialize(aText, someLevels)
    @@MAXTREASURES=10
    @text=aText
    @levels=someLevels
  end

  attr_reader :MAXTREASURES, :text
  
  def getLevels
     @levels
  end
  #Metodos abstractos
  def substractVisibleTreasure(t)
  end
  
  def substractHiddenTreasure(t)  
  end
  
  def isEmpty
  end
  
  def adjustToFitTreasureLists(v,h)
  end
  
  def to_s
  end

  
#  def substractVisibleTreasure(t)
#    if(@nVisibleTreasures==-1)
#      @specificVisibleTreasures.delete(t.getType)
#    else
#      if(@nVisibleTreasures>0)
#        @nVisibleTreasures=@nVisibleTreasures-1
#      end
#    end
#    
#  end
  
#  def substractHiddenTreasure(t)
#    if(@nHiddenTreasures==-1)
#      @specificHiddenTreasures.delete(t.getType)
#    else
#      if(@nHiddenTreasures>0)
#        @nHiddenTreasures = @nHiddenTreasures - 1
#      end
#    end
#  
#  end
   
  
#  def adjustToFitTreasureLists(v,h)
#    if(!@death)
#      if(@nHiddenTreasures!=-1 && @nVisibleTreasures!=-1)
#          a_quitarV = @nVisibleTreasures>v.size() ? v.size() : @nVisibleTreasures
#          a_quitarH = @nHiddenTreasures>h.size() ? h.size() : @nHiddenTreasures
#          out=BadConsequence.newLevelNumberOfTreasures(@text,0,a_quitarV,a_quitarH)
#      else
#        listaV=Array.new(0)
#        listaH=Array.new(0)
#        ajustadoV = Array.new(0)
#        ajustadoH = Array.new(0)
#        
#        v.each do |tesoro|
#          listaV << tesoro.getType
#        end
#        h.each do |tesoro|
#          listaH << tesoro.getType
#        end
#        
#        @specificVisibleTreasures.each do |t|
#          if((indice=listaV.index(t))!=nil)
#            ajustadoV << t
#            listaV.delete_at(indice)
#          end
#        end
#        
#        @specificHiddenTreasures.each do |t|
#          if((indice = listaH.index(t))!=nil)
#            ajustadoH << t
#            listaH.delete_at(indice)
#          end
#        end
#        out=BadConsequence.newLevelSpecificTreasures(@text,0,ajustadoV,ajustadoH)
#      end
#    else
#      out=BadConsequence.newLevelNumberOfTreasures(@text,0,v.size,h.size)
#    end
#    out
#  end
  
#  def self.newLevelNumberOfTreasures (aText, someLevels,someVisibleTreasures, someHiddenTreasures)
#    new(aText, someLevels,someVisibleTreasures, someHiddenTreasures,Array.new,Array.new,false)
#  end
#  
#  def self.newLevelSpecificTreasures (aText, someLevels,someSpecificVisibleTreasures, someSpecificHiddenTreasures)
#    new(aText, someLevels,-1,-1,someSpecificVisibleTreasures, someSpecificHiddenTreasures,false)
#  end
#  
#  def self.newDeath (aText)
#     new(aText, 0,-1,-1,Array.new,Array.new,true)
#  end
# 

#    def to_s
#      out = @text
#      if @death
#        out = out + ", Estas muerto."
#      else
#        out = out + ", pierdes #{@levels} niveles"
#        if(@nVisibleTreasures >= 0 && @nHiddenTreasures>=0 )
#        out =out + ", pierdes #{@nVisibleTreasures} tesoros visibles y #{@nHiddenTreasures} tesoros ocultos"
#        else
#        out= out + ", pierdes estos tesoros visibles:\n"
#        @specificVisibleTreasures.each do |treasure|
#          out = out + treasure.to_s + " "
#        end
#        out= out + "Y estos tesoros ocultos:\n"
#        @specificHiddenTreasures.each do |treasure|
#          out= out + treasure.to_s + " "
#        end
#        end
#    end
#    out     
#  end
end
end